src/System.Data.SqlClient/src/System/Data/SqlClient/SqlStatistics.cs

```diff
@@ -218,6 +221,100 @@ internal void UpdateStatistics()
                 _connectionTime = long.MaxValue;
             }
         }
+
+        // We subclass Dictionary to provide our own implementation of CopyTo, Keys.CopyTo, and
+        // Values.CopyTo to match the behavior of Hashtable, which is used in the full framework.
+        // Ideally this would derive from Dictionary<string, long>, but that would break compatibility
+        // with the full framework, which allows adding keys/values of any type.
+        private sealed class StatisticsDictionary : Dictionary<object, object>, IDictionary
```

Comment:
@justinvp Can you elaborate a bit more on the 

> We subclass Dictionary to provide our own implementation of CopyTo, Keys.CopyTo, and
>   // Values.CopyTo to match the behavior of Hashtable, which is used in the full framework.

Can you site some examples of the behavior difference in the comments to justify this new dictionary subclass and class Collection?